integration/kotlinx-coroutines-guava/src/ListenableFuture.kt

```diff
@@ -152,7 +152,8 @@ public fun <T> ListenableFuture<T>.asDeferred(): Deferred<T> {
     deferred.invokeOnCompletion {
         cancel(false)
     }
-    return deferred
+    // Return hides the CompletableDeferred. This should prevent casting.
+    return object : Deferred<T> by deferred {}
```

Comment:
While we are here, please fix indent on line 170 (should be 4 spaces instead of 2)